clusterloader2/pkg/measurement/common/wait_for_rc_pods.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+func init() {
+	measurement.Register("WaitForRunningRCs", createWaitForRunningRCsMeasurement)
+}
+
+func createWaitForRunningRCsMeasurement() measurement.Measurement {
+	return &waitForRunningRCsMeasurement{}
+}
+
+type waitForRunningRCsMeasurement struct{}
+
+// Execute waits until all specified RCs have all pods running or until timeout happens.
+// RCs can be specified by field and/or label selectors.
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+func (*waitForRunningRCsMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+	namespace, err := util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+	if err != nil {
+		return summaries, err
+	}
+	labelSelector, err := util.GetStringOrDefault(config.Params, "labelSelector", "")
+	if err != nil {
+		return summaries, err
+	}
+	fieldSelector, err := util.GetStringOrDefault(config.Params, "fieldSelector", "")
+	if err != nil {
+		return summaries, err
+	}
+	timeout, err := util.GetDurationOrDefault(config.Params, "timeout", 60*time.Second)
+	if err != nil {
+		return summaries, err
+	}
+
+	listOpts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	rcsList, err := config.ClientSet.CoreV1().ReplicationControllers(namespace).List(listOpts)
+	if err != nil {
+		return summaries, err
+	}
+	rcs := rcsList.Items
+	runningRCs := 0
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(5 * time.Second) {
```

Comment:
Can you make this `5 sec` value a global variable instead of hiding it within code?

Reply:
(same in wait_for_pods.go)

Reply:
I've added `globals.go` to `measurement/util` and moved this time consts there.

Reply:
I would really suggest keeping these globals together with the measurement file instead of collecting them together and putting them somewhere else. This'll avoid adding extra dependency for people while adding new measurements and it makes more sense to have the globals relevant for a file in that scope.

Reply:
Ok. Done.